### PR TITLE
Dedicated LPRewards contracts for three pairs

### PR DESCRIPTION
**Pull request by nkuba**
We created dedicated LPRewards implementations for TBTC-ETH, KEEP-ETH
and KEEP-TBTC pairs. Those were mostly created to simplify usage in the
dashboard, as we will produce separate artifact for each pair.

Depends on: #645 
Depends on: https://github.com/keep-network/keep-core/pull/2225

### CODE REVIEW

File: solidity/contracts/LPRewards.sol [hunk 1]
```diff
@@ -214,3 +221,27 @@ contract LPRewards is LPTokenWrapper, IRewardDistributionRecipient {
         }
     }
 }
+
+/// @title KEEP rewards for TBTC-ETH liquidity pool.
+contract LPRewardsTBTCETH is LPRewards {
+    constructor(KeepToken keepToken, IERC20 tbtcEthUniswapPair)
```
Comment (nkuba) [inline solidity/contracts/LPRewards.sol:227]: The first parameter could be also `IERC20`. I'm not sure what's better, use specific `KeepToken` type or more general `IERC20`?

```suggestion
    constructor(IERC20 keepToken, IERC20 tbtcEthUniswapPair)
```

Comment (pdyraga) [reply solidity/contracts/LPRewards.sol:227]: Given that those contracts are for specific pairs, I think it makes more sense to keep the code as-is.

File: solidity/contracts/LPRewards.sol [hunk 2]
```diff
@@ -100,7 +103,11 @@ contract LPTokenWrapper {
     }
 }
 
-contract LPRewards is LPTokenWrapper, IRewardDistributionRecipient {
+contract LPRewards is
+    LPTokenWrapper,
+    IRewardDistributionRecipient,
+    IStakerRewards
```
Comment (pdyraga) [inline solidity/contracts/LPRewards.sol:109]: It is no longer needed, `LPRewards` will receive tokens via `notifyRewardAmount` called by `StakingPoolRewardsEscrowBeneficiary` (see https://github.com/keep-network/keep-core/pull/2230) - it is the same mechanism as for `CurveRewards`.

File: solidity/contracts/LPRewardsEscrowBeneficiary.sol [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+
+/// @title LPRewardsEscrowBeneficiary
+/// @notice Transfer the received tokens from PhasedEscrow to a designated
+///         LPRewards contract.
+contract LPRewardsEscrowBeneficiary is StakerRewardsBeneficiary {
+    constructor(IERC20 _token, IStakerRewards _lpRewardsContract)
+        public
+        StakerRewardsBeneficiary(_token, _lpRewardsContract)
+    {}
+}
```
Comment (pdyraga) [inline solidity/contracts/LPRewardsEscrowBeneficiary.sol:13]: We do not need it, `StakingPoolRewardsEscrowBeneficiary` from https://github.com/keep-network/keep-core/pull/2230 will take care of it.

File: solidity/contracts/LPRewards.sol (hunk 2 above)
Comment (nkuba) [reply solidity/contracts/LPRewards.sol:109]: Does it mean that it has to implement `IStakingPoolRewards` instead of `IStakerRewards`?

Comment (pdyraga) [reply solidity/contracts/LPRewards.sol]: Yeah, it can implement it. I'd say it does not have to but we are also fine with implementing it ;-)

**Comment by pdyraga** [comment]
Tests are failing, I think we need to publish a new `keep-core` pre version and update the reference in dependencies.

File: solidity/test/LPRewardsTest.js [hunk 4]
```diff
@@ -222,6 +230,168 @@ describe("LPRewards", () => {
...
+      )
+
+      return beneficiary
+    }
+  })
+
+  async function mintAndApproveWrappedTokens(token, address, staker, amount) {
+    await token.mint(staker, amount)
+    await token.approve(address, amount, {from: staker})
+  }
```
Comment (pdyraga) [inline solidity/test/LPRewardsTest.js:393]: Is it used anywhere? Looks like some merge conflict resolution leftover.

**Comment by pdyraga** [comment]
The code looks good, left one comment about probably unused function. As soon as the tests start passing, I am good to merge.

**Review by pdyraga — APPROVED** [review]

**Pull request merged**